evaluator/autoware_perception_online_evaluator/src/perception_analytics_publisher_node.cpp

```diff
@@ -0,0 +1,170 @@
+      .name("all_object_count")
+      .unit("count")
+      .value(std::to_string(metrics.all_object_count)));
+
+  // object count per label
+  for (auto & label : label_list_) {
+    metrics_msg.metric_array.emplace_back(
+      tier4_metric_msgs::build<tier4_metric_msgs::msg::Metric>()
+        .name("object_count_" + convertLabelToString(label))
+        .unit("count")
+        .value(std::to_string(metrics.object_count_by_label[label])));
+  }
+
+  // max distance per label (value 0 for no object)
+  for (auto & label : label_list_) {
+    metrics_msg.metric_array.emplace_back(
+      tier4_metric_msgs::build<tier4_metric_msgs::msg::Metric>()
+        .name("max_distance_" + convertLabelToString(label))
+        .unit("m")
+        .value(std::to_string(metrics.max_distance_by_label[label])));
+  }
+
+  // latency by topic
+  metrics_msg.metric_array.emplace_back(
+    tier4_metric_msgs::build<tier4_metric_msgs::msg::Metric>()
+      .name("meas_to_tracked_latency")
+      .unit("ms")
+      .value(std::to_string(metrics.latency_by_topic_id[LATENCY_TOPIC_ID_MEAS_TO_TRACKED])));
+  metrics_msg.metric_array.emplace_back(
+    tier4_metric_msgs::build<tier4_metric_msgs::msg::Metric>()
+      .name("prediction_latency")
+      .unit("ms")
+      .value(std::to_string(metrics.latency_by_topic_id[LATENCY_TOPIC_ID_PREDICTION])));
+
+  // total latency
+  metrics_msg.metric_array.emplace_back(
+    tier4_metric_msgs::build<tier4_metric_msgs::msg::Metric>()
+      .name("total_latency")
+      .unit("ms")
+      .value(std::to_string(metrics.total_latency)));
+
+  if (!metrics_msg.metric_array.empty()) {
+    metrics_msg.stamp = now();
+    perception_analytics_pub_->publish(metrics_msg);
+  }
+}
+
+void PerceptionAnalyticsPublisherNode::onObjects(const PredictedObjects::ConstSharedPtr objects_msg)
+{
+  perception_analytics_calculator_.setPredictedObjects(*objects_msg);
+  perception_analytics_calculator_.setLatencies(latencies_);
+  publishPerceptionAnalytics();
+}
+
+rcl_interfaces::msg::SetParametersResult PerceptionAnalyticsPublisherNode::onParameter(
+  const std::vector<rclcpp::Parameter> & parameters)
+{
+  using autoware_utils::update_param;
+  auto & p = parameters_;
+
+  update_param<std::string>(
+    parameters, "meas_to_tracked_latency_topic_name", p->meas_to_tracked_latency_topic_name);
+  update_param<std::string>(
+    parameters, "prediction_latency_topic_name", p->prediction_latency_topic_name);
+
+  rcl_interfaces::msg::SetParametersResult result;
+  result.successful = true;
+  result.reason = "success";
+
+  return result;
+}
+
+void PerceptionAnalyticsPublisherNode::initParameter()
+{
+  using autoware_utils::get_or_declare_parameter;
+  auto & p = parameters_;
+
+  p->meas_to_tracked_latency_topic_name =
+    get_or_declare_parameter<std::string>(*this, "meas_to_tracked_latency_topic_name");
+  p->prediction_latency_topic_name =
+    get_or_declare_parameter<std::string>(*this, "prediction_latency_topic_name");
```

Comment:
[Just a comment]

Autoware mainly uses declare and get process. This is ROS2 style recommendation for now.

```
this->declare_parameter<std::string>("meas_to_tracked_latency_topic_name", "/default");
this->get_parameter("meas_to_tracked_latency_topic_name", p->meas_to_tracked_latency_topic_name);
```